src/main/com/yetanalytics/datasim/input.clj

```diff
@@ -12,15 +13,26 @@
             [clojure.walk :as w]
             [com.yetanalytics.datasim.util :as u]))
 
+(defn- profiles->pedges
+  [profiles]
+  (let [[templates patterns]
+        (reduce (fn [[ts ps] {:keys [templates patterns]}]
+                  (let [templates ; Exclude template relations to concepts
+                        (mapv
+                         #(select-keys % [:id :type :inScheme :prefLabel :definition])
+                         templates)]
+                    [(concat ts templates) (concat ps patterns)]))
+                [[] []]
+                profiles)]
+    (pat/get-edges (pat/create-graph templates patterns))))
+
 (s/def ::profiles
-  (s/every (s/and (s/conformer (fn [x]
-                                 ;; Only operate on maps, or invalidator tests
-                                 ;; get an unhandled error
-                                 (if (map? x)
-                                   (u/remove-nil-vals x)
-                                   x)))
-                  ::ps/profile) :min-count 1
-           :into []))
+  (s/and
+   (s/every ::ps/profile :min-count 1 :into [])
+   ;; Validate that all edges with a Pattern src ends up at a Pattern or
+   ;; Template dest that is also in the profile cosmos.
+   (s/conformer profiles->pedges)
```

Comment:
Yep. Now that you pointed it out though, I realize I could drop that line because `::pat/valid-edges` doesn't care about where these optional IRIs point to (`pan/validate-profile` did, which is why I had that line in the first place).

Reply:
Cool. I don't even know if being lossless about input profiles is a requirement in any way, just curious